Common/cpp/SharedItems/Shareables.cpp

```diff
@@ -66,6 +66,9 @@ jsi::Value makeShareableClone(
         shareable = std::make_shared<ShareableArray>(rt, object.asArray(rt));
       }
     } else if (object.isHostObject(rt)) {
+      if (object.isHostObject<ShareableJSRef>(rt)) {
+        return object;
+      }
```

Comment:
We should never call `makeShareableClone` with an object that is already an `ShareableJSRef`. This would be an error in our logic. Let's simply fail in such case.
```suggestion
      assert(!object.isHostObject<ShareableJSRef>(rt) && "[Reanimated] Provided value is already an instance of ShareableJSRef.");
```